src/addons/dragAndDrop/DateContentRowWrapper.js

```diff
@@ -42,21 +45,69 @@ class DateContentRowWrapper extends Component {
     this.setState({ ...next });
   }
 
-  _posEq = (a, b) =>
-    a.span === b.span && a.left === b.left && a.right === b.right && a.level === b.level;
+  _posEq = (a, b) => a.left === b.left && a.level === b.level;
 
   handleSegmentDrag = drag => {
     this.setState({ drag });
   };
 
-  handleSegmentHover = (hover, hoverData) => {
-    const { drag } = this.state;
+  handleBackgroundCellHoverExit = () => {
+    const props = withLevels(this.props);
+    this.setState({ ...props, drag: null });
+  };
+
+  handleSegmentHover = ({ position: hover, data: hoverData }, dragEvent) => {
+    const { type: dragEventType, data: dragData, ...dragRest } = dragEvent;
+    let { drag } = this.state;
+    let { events } = this.props;
+    if (!drag && dragEventType === 'outsideEvent') {
+      // update position based on hover
+      const { position: { span } } = dragRest;
+      const dragPos = { ...hover, span, level: hover.level + 1 };
+      const { id: eventTemplateId, eventTemplateId: id, ...dragDataRest } = dragData;
+
+      // calculate start and end
+      const data = {
```

Comment:
none of this should be in here :( but ran out of time to think of a proper solution